### PR TITLE
Matrix chain multiplication in C language.

### LINKED ISSUE #1794: Matrix Chain Multiplication in C
I want to add matrix chain multiplication in C as it is not available, for gssoc 2020. @asha15 @singh-shreya6 @raghav-dalmia kindly assign it to me.

### PR DESCRIPTION
**Pull request by NVS16**
Fixes #1794 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Matrix chain multiplication in C language added

#### Languages Used:
- C

#### Files Added:
- Matrix_Chain_Multiplication.c

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by MastersAbh** [comment]
The issue has still not been assigned to you. Please do not create PR before that.

File: Matrix_Chain_Multiplication/Matrix_Chain_Multiplication.c [hunk 1]
```diff
@@ -0,0 +1,99 @@
+                                                                     /*DESCRIPTION*/
+
+/*Let us define the problem as trying determine the order of multiplication M1*M2*M3*...*Mn using parentheses to denote the order.
```
Comment (singh-shreya6) [inline Matrix_Chain_Multiplication/Matrix_Chain_Multiplication.c]: Please shorten your problem description. Such a huge comment is not needed as part of code. A Readme file separately can have such great details. The Problem description should be 1 brief paragraph containing 2-3 lines.

File: Matrix_Chain_Multiplication/Matrix_Chain_Multiplication.c [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    int i, j, k, L, q;
+  
+    // cost is zero when multiplying one matrix.
+    for (i=1; i<n; i++) 
+        m[i][i] = 0; 
+  
+    // L is chain length. 
+    for (L = 2; L < n; L++) 
+    { 
+        for (i = 1; i< n-L+1; i++)
```
Comment (singh-shreya6) [inline Matrix_Chain_Multiplication/Matrix_Chain_Multiplication.c]: Please give space around operators

File: Matrix_Chain_Multiplication/Matrix_Chain_Multiplication.c [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+#include <limits.h>
+
+int MCM(int *p, int n) 
+{
+    int m[n][n]; 
+  
+    int i, j, k, L, q;
+  
+    // cost is zero when multiplying one matrix.
+    for (i=1; i<n; i++)
```
Comment (singh-shreya6) [inline Matrix_Chain_Multiplication/Matrix_Chain_Multiplication.c]: Space around operators

**Comment by NVS16** [comment]
Hey @singh-shreya6 
I have added the requested changes. You can review them.

File: Matrix_Chain_Multiplication/Matrix_Chain_Multiplication.c [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            { 
+                // q = cost/scalar multiplications 
+                q = m[i][k] + m[k + 1][j] + p[i - 1] * p[k] * p[j]; 
+                if (q < m[i][j]) 
+                    m[i][j] = q; 
+            } 
+        } 
+    } 
+  
+    return m[1][n-1];
```
Comment (singh-shreya6) [inline Matrix_Chain_Multiplication/Matrix_Chain_Multiplication.c]: please add space around operators

File: Matrix_Chain_Multiplication/Matrix_Chain_Multiplication.c [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    for (int i = 0; i < n + 1; i++) {
+        scanf("%d", &P[i]);
+    }
+    
+    int size = sizeof(P) / sizeof(int);
+
+    printf("The minimum number of scalar multiplications: %d\n", MCM(P, size));
+}
+
+                                                                    /*OUTPUT DESCRIPTION*/
```
Comment (singh-shreya6) [inline Matrix_Chain_Multiplication/Matrix_Chain_Multiplication.c]: It is right aligned a bit too much. Make it center aligned.

File: Matrix_Chain_Multiplication/Matrix_Chain_Multiplication.c [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+
+         This is how table[][] matrix will look like
+
+                0 6000 18000 30000
+                0 0    24000 48000
+                0 0    0     36000
+                0 0    0     0
+
+        ,where table[1][size-1]=30000 is the answer.
```
Comment (singh-shreya6) [inline Matrix_Chain_Multiplication/Matrix_Chain_Multiplication.c]: Remove the comma here.

**Comment by NVS16** [comment]
Have made the requested changes.

File: Matrix_Chain_Multiplication/Matrix_Chain_Multiplication.c [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+                /* OUTPUT DESCRIPTION */
+/*
+              P = {10, 20, 30, 40, 30},
+              dimensions of matrix [1] = 10X20,
+              dimensions of matrix [2] = 20X30,
+              dimensions of matrix [3] = 30X40,
+              dimensions of matrix [4] = 40X30
+
+
+         This is how table[][] matrix will look like
```
Comment (singh-shreya6) [inline Matrix_Chain_Multiplication/Matrix_Chain_Multiplication.c]: what does table[][] signify here?
If not used in the code, can you remove this from explanation ? 
I'll approve after that. Rest all looks good to me.

**Comment by NVS16** [comment]
I gave meaningful names to variables. 
Changed m[][] -> table[][]
Changed P[] -> dims[]

**Pull request merged**